src/lib/common/topology_sampler.cpp

```diff
@@ -149,12 +220,255 @@ static void topology_timer_cb(void *arg) {
  * @return none
  */
 static void topology_timer_handler(TimerHandle_t tmr) {
-    (void) tmr;
-    timer_callback_handler_send_cb(topology_timer_cb, NULL, 0);
+  (void)tmr;
+  timer_callback_handler_send_cb(topology_timer_cb, NULL, 0);
 }
 
+/*!
+ * @brief Callback function for sys info reply, sends to queue for processing.
+ * 
+ * @param[in] ack True if the message was acknowledged, false otherwise.
+ * @param[in] msg_id The message id.
+ * @param[in] service_strlen The length of the service string.
+ * @param[in] service The service string.
+ * @param[in] reply_len The length of the reply data.
+ * @param[in] reply_data The reply data.
+ * @return True if the request was handled, false otherwise.
+ */
+static bool sys_info_reply_cb(bool ack, uint32_t msg_id, size_t service_strlen,
+                              const char *service, size_t reply_len, uint8_t *reply_data) {
+  if (ack) {
+    SysInfoSvcReplyMsg::Data reply = {0, 0, 0, 0, NULL};
+    if (SysInfoSvcReplyMsg::decode(reply, reply_data, reply_len) != CborNoError) {
+      printf("Failed to decode sys info reply\n");
+      return false;
+    }
+    printf("Service: %.*s\n", service_strlen, service);
+    printf("Reply: %" PRIu32 "\n", msg_id);
+    printf(" * Node id: %" PRIx64 "\n", reply.node_id);
+    printf(" * Git SHA: 0x%08" PRIx32 "\n", reply.git_sha);
+    printf(" * Sys config CRC: 0x%08" PRIx32 "\n", reply.sys_config_crc);
+    printf(" * App name: %.*s\n", reply.app_name_strlen, reply.app_name);
+    if (xQueueSend(_sys_info_queue, &reply, 0) != pdTRUE) {
+      printf("Failed to send sys info\n");
+    }
```

Comment:
The call to `decode` here (as well as on line 277 where I won't repeat the same comment) allocates memory for `app_name`. It took me a while to follow these through the queues and functions to determine that they were being freed.

So, first, some comments could be helpful there, like on the line before the `xQueueSend`, comment that `app_name` will be freed on the receiving side, and on the line before the subsequent encode call, say that the encoding function frees `app_name`.

Second, allocated memory needs to be freed in case of errors, for example around line 301 if the queue send fails, and probably more cases like that.